File: core/src/com/colorflow/ring/PolarListener.java
package com.colorflow.ring;

import com.badlogic.gdx.InputAdapter;
import com.colorflow.utility.Position;

class PolarListener extends InputAdapter implements RingListener {

    private Ring ring;
    private Position actual, last;

    PolarListener(Ring ring) {
        this.ring = ring;
        this.actual = new Position.Pixel(0, 0);
        this.last = new Position.Pixel(0, 0);
    }

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) {
        last.setX(screenX);
        last.setY(screenY);
        return false;
    }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) {
        actual.setX(screenX); actual.setY(screenY);
        ring.setRotation(ring.getRotation() + (last.getAngleRadial() - actual.getAngleRadial()) * ring.getSensibility());
        last.setX(screenX); last.setY(screenY);
        return false;
    }

    @Override
    public void onRingAct() {

    }
}

File: core/src/com/colorflow/play/ring/PolarListener.kt
package com.colorflow.play.ring

import com.badlogic.gdx.InputAdapter
import com.colorflow.utility.Position

internal class PolarListener(private val ring: Ring) : InputAdapter(), RingListener {
    private val actual: Position
    private val last: Position

    init {
        this.actual = Position.Pixel(0f, 0f)
        this.last = Position.Pixel(0f, 0f)
    }

    override fun touchDown(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean {
        last.x = screenX.toFloat()
        last.y = screenY.toFloat()
        return false
    }

    override fun touchDragged(screenX: Int, screenY: Int, pointer: Int): Boolean {
        actual.x = screenX.toFloat()
        actual.y = screenY.toFloat()
        ring.rotation = ring.rotation + (last.angleRadial - actual.angleRadial) * ring.sensibility
        last.x = screenX.toFloat()
        last.y = screenY.toFloat()
        return false
    }

    override fun onRingAct() {

    }
}